### ISSUE TITLE
create stack on label provider process spinning

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#7158]
**Rancher Version:**
v1.2.1

In a scenario where k8s services are spinning, this process seems to be prioritized, rescheduled and never succeeds... leading to thread starvation.

**Comment by deniseschannon** [comment]
Available on the rancher-catalog branch `1.2.2-rc2` in kubernetes-agent:v0.4.7 in 

in both kubernetes version:
kubernetes:v1.4.6-rancher2-4
kubernetes:v1.5.1-rancher1-2

**Comment by galal-hussein** [comment]
Tested with Rancher version v1.2.2-rc2 and catalog branch 1.2.2-rc2 with kubernetes-agent:v0.4.7

The steps to validate the issue:

- Run a 50 rc with scale 2 with /bin/false as an entrypoint to ensure crashloop situation
- Run a replication controller with 100 pods with a valid image.

At this point some pods were unable to start, after leaving the setup for a 1 hour all the 100 pods recovered to a running state.

- Run 20 more pods.

All 20 pods ran successfully and the containerevent.create was 38:

```
mysql> select count(process_name), process_name from process_instance where end_time is null group by process_name;
+---------------------+-----------------------------+
| count(process_name) | process_name                |
+---------------------+-----------------------------+
|                  38 | containerevent.create       |
|                  10 | image.create                |
|                  34 | instance.create             |
|                   9 | instance.remove             |
|                   3 | instance.stop               |
|                  14 | mount.create                |
|                  15 | mount.deactivate            |
|                  41 | volume.activate             |
|                   8 | volume.create               |
|                  15 | volume.deactivate           |
|                  18 | volumestoragepoolmap.purge  |
|                   6 | volumestoragepoolmap.remove |
+---------------------+-----------------------------+
```
  
I didn't notice any errors that were seen before in the cattle debug log.
